pkg/scheduler/framework/plugins/interpodaffinity/scoring.go

```diff
@@ -142,6 +142,15 @@ func (pl *InterPodAffinity) PreScore(
 	hasPreferredAffinityConstraints := affinity != nil && affinity.PodAffinity != nil && len(affinity.PodAffinity.PreferredDuringSchedulingIgnoredDuringExecution) > 0
 	hasPreferredAntiAffinityConstraints := affinity != nil && affinity.PodAntiAffinity != nil && len(affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution) > 0
 
+	// Optionally ignore calculating preferences of existing pods' affinity rules
+	// if the incoming pod has no inter-pod affinities.
+	if pl.args.IgnorePreferredTermsOfExistingPods && !hasPreferredAffinityConstraints && !hasPreferredAntiAffinityConstraints {
+		cycleState.Write(preScoreStateKey, &preScoreState{
+			topologyScore: make(map[string]map[string]int64),
+		})
+		return nil
```

Comment:
@alculquicondor @Huang-Wei are we going to allow returning `Skip` for PreScore extensions to indicate that Score shouldn't run for the same plugin?

Reply:
that sounds like a better contract, compared to plumbing the placeholder and checking if it's empty in Score.

Reply:
yes, we should have support for Skip in Score as well, but the work didn't start yet.